### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by Rahgooy**
refactoring

### CODE REVIEW

**Comment by Rahgooy** [comment]
@kordjamshidi please have a look and merge if it's OK.

**Comment by Rahgooy** [comment]
I don't know why the slack integration was removed!

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+  val segmentList = imageReader.getSegmentList.filter(s => imageList.exists(_.getId == s.getAssociatedImageID))
+  val imageRelationList = imageReader.getImageRelationList.filter(r => imageList.exists(_.getId == r.getImageId))
 
   val tokens = sentenceList.flatMap(sentenceToTokenGenerating)
-  setTokenRolesFromXML(tokens, proportion)
-  val trCandidates = getTrajectorCandidates(tokens, true)
-  val lmCandidates = getLandmarkCandidates(tokens, true)
-  val spCandidates = getIndicatorCandidates(tokens, true)
+  xmlReader.setTokenRoles(tokens)
+  val trCandidates = MultiModalCandidateGenerator.getTrajectorCandidates(tokens, true)
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala]: why multimodal generator here? isn't it only based on text in here?

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+  val lmCandidates = MultiModalCandidateGenerator.getLandmarkCandidates(tokens, true)
+  val spCandidates = MultiModalCandidateGenerator.getIndicatorCandidates(tokens, true)
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala]: same as above.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala (hunk 1 above)
Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/DataExplorationApp.scala]: ah, right

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/MultiModalXmlReader.scala [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import edu.illinois.cs.cogcomp.saulexamples.nlp.Xml.NlpXmlReader
+import edu.illinois.cs.cogcomp.saulexamples.nlp.XmlMatchings
+
+import scala.collection.JavaConversions._
+
+/**
+  * Created by taher on 2017-02-28.
+  */
+class MultiModalXmlReader(dataDir: String, proportion: DataProportion) {
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/MultiModalXmlReader.scala]: I guess this is only based on text, so maybe the prefix multimodel here is redundant?

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/ReportHelper.scala [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+import scala.util.control.Breaks.{break, breakable}
+
+/**
+  * Created by taher on 2017-02-28.
+  */
+object ReportHelper {
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/ReportHelper.scala:18]: nice

**Comment by kordjamshidi** [comment]
good to merge, for me. Thanks!

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/MultiModalXmlReader.scala (hunk 3 above)
Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/MultiModalXmlReader.scala]: is `XmlReaderHelper` good?

Comment (kordjamshidi) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/Helpers/MultiModalXmlReader.scala]: yes, I think so.

**Pull request merged**